### PR TITLE
[CodeGen] Increase NumVisited limit in TwoAddressInstructionPass to 64

### PR DESCRIPTION
**Pull request by AZero13**
Now that hardware has progressed, we can greatly increase the limit to something larger, allowing room for more optimization.

### CODE REVIEW

**Comment by RKSimon** [comment]
@AtariDreams please can you work with @nikic to investigate the effect on compile time : https://llvm-compile-time-tracker.com/

**Comment by AtariDreams** [comment]
> @AtariDreams please can you work with @nikic to investigate the effect on compile time : https://llvm-compile-time-tracker.com/

Maybe I should find out when the returns diminish enough and set that as the limit
Across 32 changed tests, I have a delta of around 250 instructions saved.

File: llvm/lib/CodeGen/TwoAddressInstructionPass.cpp [hunk 1]
```diff
@@ -918,11 +918,11 @@ bool TwoAddressInstructionPass::rescheduleMIBelowKill(
   MachineBasicBlock::iterator KillPos = KillMI;
   ++KillPos;
   for (MachineInstr &OtherMI : make_range(End, KillPos)) {
+    if (NumVisited >= 18) // FIXME: Arbitrary limit to reduce compile time cost.
+      return false;
```
Comment (nikic) [inline llvm/lib/CodeGen/TwoAddressInstructionPass.cpp:922]: This check must come after isDebug, otherwise you will break debug invariance.

**Comment by nikic** [comment]
Compile-time looks fine.

**Comment by david-arm** [comment]
The AArch64 changes seem reasonable to me. Perhaps it's worth updating the title and/or commit message to provide more detail, i.e. `[CodeGen] Increase NumVisited limit in TwoAddressInstructionPass to 18`.

**Comment by nikic** [comment]
> ⚠️ We detected that you are using a GitHub private e-mail address to contribute to the repo. Please turn off [Keep my email addresses private](https://github.com/settings/emails) setting in your account.

Please resolve this warning.

**Comment by goldsteinn** [comment]
Should the bound maybe be a `cl::opt`?

**Comment by AtariDreams** [comment]
> Should the bound maybe be a `cl::opt`?

I don't think that is needed.

**Comment by AtariDreams** [comment]
@topperc Thoughts?

**Comment by efriedma-quic** [comment]
I'm skeptical it's a good idea to remove the limit completely.  The reason we have thresholds like this is that it allows us to use simple algorithms that would otherwise be O(n^2) or worse.  Usually what happens is that the code appears to work fine on common benchmarks, but then someone files a bug report saying the compiler times out in specific cases.

**Comment by nikic** [comment]
The previously version of this PR that just raised the limits a bit looked fine to me.

> > Should the bound maybe be a `cl::opt`?
> 
> I don't think that is needed.

It's indeed not needed, but it's pretty common to use a cl::opt for such cutoffs, so it's easier to test different value for them.

**Pull request closed without merging**